src/main/java/controllers/MovementGameAction.java

```diff
@@ -0,0 +1,32 @@
+package controllers;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import cs361.battleships.models.Ship;
+import cs361.battleships.models.Submarine;
+import cs361.battleships.models.Game;
+
+public class MovementGameAction {
+
+    @JsonProperty private Game game;
+    @JsonProperty private String direction;
+
+    public Game getGame() {
+        return game;
+    }
+
+    public void replaceSubmarine()
+    {
+       for(int i = 0; i < this.getGame().getPlayersBoard().getShips().size(); i++)
```

Comment:
Yeah you were right, this ended up really ugly. I had problems accessing the submarine how I wanted to as well. I think if we were continuing to develop this, we could refactor quite a bit.

In fact, I believe that Sprint 3 should've just been refactoring. Adding features alongside refactoring while having a requirement to not modify many function signatures really tied our hands behind our back...